### PR TITLE
terminal codes mapping for nat gateway

### PR DESCRIPTION
**Pull request by LikithaVemulapalli**
Issue #, if available: [1362](https://github.com/aws-controllers-k8s/community/issues/1362)

Description of changes:

- Mapped terminal codes for **Nat Gateway** resource.
- Added integration tests for the mapped terminal codes.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/e2e/tests/test_nat_gateway.py [hunk 1]
```diff
@@ -117,4 +117,111 @@ def test_create_delete(self, standard_elastic_address, ec2_client):
...
+
+    def test_terminal_condition_missing_parameter(self):
+        test_resource_values = REPLACEMENT_VALUES.copy()
+        resource_name = random_suffix_name("nat-gateway-fail-3", 24)
+        test_vpc = get_bootstrap_resources().SharedTestVPC
+        subnet_id = test_vpc.public_subnets.subnet_ids[0]
+
+        test_resource_values["NAT_GATEWAY_NAME"] = resource_name
+        test_resource_values["SUBNET_ID"] = subnet_id
+        test_resource_values["ALLOCATION_ID"] = ""
```
Comment (brycahta) [inline test/e2e/tests/test_nat_gateway.py:201]: nit: would you mind adding a comment saying `allocation_id` is required for public nat gateway only?

**Review by brycahta — COMMENTED** [review]
Looks good 👍 

any test failures might require more deep dive per Nick's comment in slack channels. related PR:  https://github.com/aws-controllers-k8s/community/pull/1377

File: test/e2e/tests/test_nat_gateway.py (hunk 1 above)
Comment (LikithaVemulapalli) [reply test/e2e/tests/test_nat_gateway.py:201]: Will add a comment in this test 👍

**Review by brycahta — COMMENTED** [review]
/retest

**Comment by brycahta** [comment]
@LikithaVemulapalli given the ongoing changes and retests, this could be in a weird state. Would you mind re-opening this PR? I think that'd be the quickest solution.

**Comment by LikithaVemulapalli** [comment]
> @LikithaVemulapalli given the ongoing changes and retests, this could be in a weird state. Would you mind re-opening this PR? I think that'd be the quickest solution.

Yes, I was thinking the same, will re-open PR.

**Pull request closed without merging**